run-tests.py

```diff
@@ -68,14 +68,19 @@ def dev(args, remain):
     tag = get_tag(args.image, args.salt)
     if not image_exists(tag):
         _build(args.image, args.salt)
-    subprocess.call([
-        "docker", "run", "--rm", "-it",
+    cmd = [
+        "docker", "run", "-d",
         "-v", "{0}/test/minion.conf:/etc/salt/minion.d/minion.conf".format(BASEDIR),
         "-v", "{0}/test/salt:/srv/salt".format(BASEDIR),
         "-v", "{0}/test/pillar:/srv/pillar".format(BASEDIR),
         "-v", "{0}/{1}:/srv/formula/{1}".format(BASEDIR, _formula),
-        tag, "/bin/bash",
-    ])
+    ]
+    cmd.append(tag)
+    docker_id = subprocess.check_output(cmd).strip()
+    try:
+        subprocess.call(["docker", "exec", "-it", docker_id, "/bin/bash"])
+    finally:
+        subprocess.call(["docker", "rm", "-f", docker_id])
```

Comment:
I don't get the point of all this, nor how this is related to what's mentioned in the commit message. Perhaps add a comment to explain?